enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,268 @@
+reviewers:
+  - "@tremes"
+  - "@inecas"
+approvers:
+  - TBD
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+creation-date: 2022-02-14
+last-updated: 2022-02-17
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/CCXDEV-6653
+see-also:
+replaces:
+superseded-by:
+---
+
+# Insights Operator Prometheus Alerts for Insights Recommendations
+
+## Summary
+
+Insights recommendations are triggered based on data in the archives
+periodically uploaded by the Insights Operator. A report about these
+recommendations is available through the Insights Smart Proxy, and it is
+currently fetched by the Insights Operator after each successful archive
+upload in order for the Insights Operator to update Prometheus metrics
+containing the number of currently active recommendations in each severity
+category.
+
+This enhancement aims to extend this functionality by not only making the
+number of recommendations available, but also providing specific information
+about each of the recommendations via Prometheus alerts. The currently
+suggested approach expects one info-level alert per active Insights
+recommendation, containing its ID, human-readable name, severity, and a link
+to detailed information about the recommendation and its remediation steps.
+
+## Motivation
+
+Users are currently not being immediately notified about newly active Insights
+recommendations. If they do not check the Insights tab periodically, these
+recommendations may go unnoticed for a long time during which the underlying
+issue escalates instead of being resolved. Making the recommendations available
+through the Prometheus alerts makes it easier for users to define how they want
+to receive information about newly active recommendations (via an email, Slack
+message, etc.).
+
+### Goals
+
+- New Prometheus metric with labels containing aforementioned properties of
+  the recommendation that will indicate if the recommendation is currently active.
+- New Prometheus alert based on this metric, defined using a template that fills
+  the information from labels into the description of the alert.
+- The user should be able to find out when was a certain recommendation active
+  by querying the Prometheus metric in the OCP console.
+- Configuration option allowing the user to disable/enable this functionality.
+  Since this feature consumes resources on both sides (cluster and Smart Proxy),
+  it should only be enabled on clusters where the user actually wants to use it. (It should be disabled by default, unless the user explicitly enables it.)
+- The enabling should be done using the same approach as the rest of the
+  Insights Operator configuration (i.e., currently using the `support` secret).
+
+### Non-Goals
+
+- Insights Operator fetching additional data, from the Smart Proxy or any other source.
```

Comment:
It's a feature added for the users, not for us, so the motivation to enable it would be that they want to use it. Enabling it by default does not make sense for us because most users will likely not be interested in it, and we wouldn't want to spam them with alerts they don't care about, but most importantly because this feature costs us additional resources (fetching the Insights report adds load onto the Smart Proxy), which would be wasted if the user didn't care about the alerts and didn't bother disabling the feature. I see not benefit from enabling this by default. We expect that no one would use this without configuring some notifications via Alertmanager, so there will always be some configuration necessary to actually utilize the feature for its intended purpose.

I am not really sure what the difference between the two options is, but it is saying that the Insights Operator will be fetching the exact same resources it is fetching already. We are already getting the full report from the Smart Proxy, and will continue to do that. The only difference is that we will extract more data from it than we currently do. The reason I added this point was to clarify that we do not want/need to access a new endpoint, and neither does the Smart Proxy need to expose additional information in the reports it returns. The Smart Proxy will not have to be modified in any way, and IO will be making the same exact request, and will expect the same exact data.